lms/djangoapps/grades/api/tests/test_views.py

```diff
@@ -141,8 +142,14 @@ def test_self_get_grade(self):
         """
         Test that a user can successfully request her own grade.
         """
-        resp = self.client.get(self.get_url(self.student.username))
-        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        with check_mongo_calls(6):
+            resp = self.client.get(self.get_url(self.student.username))
+            self.assertEqual(resp.status_code, status.HTTP_200_OK)
+
+        # redo with block structure now in the cache
+        with check_mongo_calls(3):
```

Comment:
**Note to reviewers:**  Moving the creation of the test course into a bulk_operation significantly improves the setup time for these tests (since a publish-signal isn't sent for each of the 75+ blocks created).